benchmark/smallbank/protos/smallbank.proto

```diff
@@ -0,0 +1,129 @@
+/**
+* Copyright 2017 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+syntax = "proto3";
+option java_multiple_files = true;
+
+option go_package = "smallbank_pb2";
+
+message Account {
+    // Customer ID
+    uint32 customer_id = 1;
+
+    // Customer Name
+    string customer_name = 2;
+
+    // Savings Balance (in cents to avoid float)
+    uint32 savings_balance = 3;
+
+    // Checking Balance (in cents to avoid float)
+    uint32 checking_balance = 4;
+}
+
+message SmallbankTransactionPayload {
+    message CreateAccountTransactionData {
+        // The CreateAccountTransaction creates an account
+
+        // Customer ID
+        uint32 customer_id = 1;
+
+        // Customer Name
+        string customer_name = 2;
+
+        // Initial Savings Balance (in cents to avoid float)
+        uint32 initial_savings_balance = 3;
+
+        // Initial Checking Balance (in cents to avoid float)
+        uint32 initial_checking_balance = 4;
+    }
+
+    message DepositCheckingTransactionData {
+        // The DepositCheckingTransction adds an amount to the customer's
+        // checking account.
+
+        // Customer ID
+        uint32 customer_id = 1;
+
+        // Amount
+        uint32 amount = 2;
+    }
+
+    message WriteCheckTransactionData {
```

Comment:
Anybody could send a WriteCheckTransaction to remove an amount from the account? not must the owner of the account?

Reply:
It is up to application to validate authentication and authorization, since we are testing sawtooth functionality we are doing minimal validation in transaction processor to check whether sufficient balance , whether account exists or not etc.

Reply:
I got it.